README.md

```diff
@@ -34,7 +35,50 @@ var User = Arrow.Model.extend('user', {
 			attachment_content: { name: 'content', type: String }
 		},
 		connector: 'appc.mongo'
-	});
+	})
+```
+
+#### Selecting Fields
+
+To select precise fields from a data source, for the majority of cases making a model with fields with keys which match those of a specified model, and specifying the correct type is enough:
+
+```javascript
+Arrow.Model.extend('simpleUser', {
+	fields: {
+			first_name: { type: String, model: 'user' },
+			last_name: { type: String, model: 'user' }
+	},
+	connector: 'appc.composite'
+})
+```
+
+In some cases this isn't precise enough. Using a field of type: Array or Object can result in "Selecting Whole Models Instead of Fields" or "Joining with Multiple Children" as described below. To make sure that the response is always as expected, when selecting individual fields from models, always specify a "name" property, even when it's identical to the field key.
```

Comment:
"expected" behaviour is down to the user. The idea is the user may specify
```
roles: { type: Array, model: 'user' }
```
Given the format of first_name and last_name, the user would expect to get back an array which is the field "roles" from the model "user". Instead they will get back an array of users (join as array).

This is the reason why i'm recommending that the user specifies "name" for all fields which are specifically individual fields of a model, even if it isn't necessary in the case of String, Number etc...